pkgs/development/libraries/mkldnn/default.nix

```
@@ -0,0 +1,36 @@
+{ stdenv, fetchFromGitHub, cmake, substituteAll, bash, autoPatchelfHook }:
+
+stdenv.mkDerivation rec {
+  pname = "mkldnn";
+  version = "1.0.2";
+
+  src = fetchFromGitHub {
+    owner = "intel";
+    repo = "mkl-dnn";
+    rev = "v${version}";
+    sha256 = "0lf14a4j511pydyz78wwbl6hifbq86m3dy119iss9cvz8zzrdyvw";
+  };
+
+  patches = [ (substituteAll {
+    src = ./bash-to-sh.patch;
+    inherit bash;
+  }) ];
+
+  nativeBuildInputs = [ cmake ];
+
+  doCheck = true;
+
+  checkInputs = [ autoPatchelfHook ];
```

I would still add a comment stating why this is necessary
```suggestion
  # the test executables aren't able to find libmkldnn.so through LD_LIBRARY_PATH, so patch them.
  checkInputs = [ autoPatchelfHook ];
```